src/components/utils/test/scope_guard_test.cc

```diff
@@ -90,7 +90,7 @@ TEST(ScopeGuardTest, CallObjectFunctionWithParam) {
 TEST(ScopeGuardTest, DismissCallFreeFunctionWithParam) {
   {
     call_with_param_count = 0;
-    char* ptr = new char;
+    char* ptr = const_cast<char*>("dealloc is not called");
```

Comment:
@SNiukalov Can you please explain this magic?

Reply:
@AByzhynar 
In this case, we check that the CallFreeFunction will never be called and the memory will not be freed.
But we pass the pointer from the heap and make a memory leak.
I change this behavior and leave a reminder with text: dealloc not called

Reply:
@SNiukalov Can you use just `const char *` without casting ?